### DISCUSSION TITLE
Consider Adopting Conventional Commit Names for Better Version Control Understanding

### DISCUSSION THREAD

**Discussion by Niravanaa** [NathanGrenier/SOEN-341#62] [Ideas]
I'd like to suggest embracing the use of Conventional Commit Names outlined in [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0) for our project. This approach standardizes commit messages into a simple format that follows clear rules, making it easier for all team members to understand the changes introduced in each commit.

This practice would streamline our development process, enhance communication, and assist in maintaining a more organized and coherent commit history.

```
<type>[optional scope]: <description>

[optional body]

[optional footer(s)]
```

**Comment by NathanGrenier** [comment]
I was already kind of doing that on the branch level (with: feat/, fix/, doc/). I've never looked too into a convention for commits but it's probably worth a read. 

We could make a wiki page summarizing the conventions we want to apply for this project.

**Comment by briantkatch** [comment]
I prefer imperative commit messages with minimum extra data in the subject line, and the 50/72 rule for line length: https://cbea.ms/git-commit/#imperative

@NathanGrenier git-flow specifies the full word 'feature/' as the prefix for a feature branch name, so let's use that or document the short version in a config file.

**Comment by NathanGrenier** [comment]
I created a [PR](#63) adding an optional dev tool to create commit messages that adhere to conventional commit's standards. You aren't forced to use the tool but should for important commits.
